File: 3233/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(nonSpecialCount(4, 16))
}

func nonSpecialCount(l int, r int) int {
	n := int(math.Sqrt(float64(r)))
	v := make([]int, n+1)
	res := r - l + 1
	for i := 2; i <= n; i++ {
		if v[i] == 0 {
			if i * i >= l && i * i <= r {
				res--
			}
			for j := i * 2; j <= n; j += i {
				v[j] = 1
			}
		}
	}
	return res
}

File: 3233/main.rs
struct Solution {}

impl Solution {
    pub fn non_special_count(l: i32, r: i32) -> i32 {
        let n = (r as f64).sqrt() as usize;
        let mut v = vec![0; n+1];
        let mut res = r - l + 1;
        for i in 2..=n {
            if v[i] == 0 {
                let square = (i * i) as i32;
                if square >= l && square <= r {
                    res -= 1;
                }
                for j in (i * 2..=n).step_by(i) {
                    v[j] = 1;
                }
            }
        }
        res
    }
}

fn main() {
    println!("{}", Solution::non_special_count(4, 16));
}